src/Pim/Component/Catalog/Model/AbstractProduct.php

```diff
@@ -667,4 +671,24 @@ public function getReference()
     {
         return $this->getIdentifier();
     }
+
+    /**
+     * @return ArrayCollection
+     */
+    public function getUniqueData()
```

Comment:
As the word "data" takes no "s", plus here we get the "unique" data, I think it's confusing. Maybe it can be nice to explicitly tell it's a collection, exceptionally:
```
getUniqueDataCollection()
```